lib/cdo/i18n_backend.rb

```diff
@@ -164,12 +164,88 @@ def translate(locale, key, options = ::I18n::EMPTY_HASH)
       end
     end
 
-    class SimpleBackend < ::I18n::Backend::Simple
+    module Plugins
       include SmartTranslate
       include MarkdownTranslate
       include SafeInterpolation
       include I18nStringUrlTrackerPlugin
     end
+
+    class SimpleBackend < ::I18n::Backend::Simple
+      include Plugins
+    end
+
+    # The class extends the `LazyLoadable` backend from the i18n gem to support CDO i18n files:
+    # - Modified the `LocaleExtractor` class to support filenames that include both a prefix and the locale (e.g., "courses.en-US.yml").
+    # - Modified the `filenames_for_current_locale` method to load files for the current locale and its fallbacks, (e.g, "en" for "en-US" and vice versa)
+    # - Modified the `eager_load!` method to ignore eager loading when lazy loading is enabled.
+    # - Prevented the `assert_file_named_correctly!` method from raising an irrelevant error regarding file locale mismatches.
+    # - Implemented a mechanism to avoid reloading already loaded translation files.
+    # https://github.com/ruby-i18n/i18n/blob/v1.12.0/lib/i18n/backend/lazy_loadable.rb
+    class LazyLoadableBackend < ::I18n::Backend::LazyLoadable
+      include Plugins
+
+      LOCALES_MAPPING = YAML.load_file(CDO.dir('dashboard/config/locales.yml')).each_with_object({}) do |(k, v), locales|
+        locales[k.to_sym] = v.to_sym if v.is_a?(String)
+      end.freeze
+
+      # The original method has been modified to works on i18n files named with a prefix and the locale,
+      # like "common.en.yml" or "common.en-US.json" and not only on files named with the locale, like "en-US.json".
+      # https://github.com/ruby-i18n/i18n/blob/v1.12.0/lib/i18n/backend/lazy_loadable.rb#L55-L63
```

Comment:
Very nice touch linking to a tagged version rather than just an arbitrary commit!